### PR TITLE
Draft Feature: Add SQLite

### PR DESCRIPTION
**Pull request by networkfusion**
## Motivation and Context
A long time feature request is to add SQLite support. This PR adds the skeleton for such a request and is backed up by the managed layer in https://github.com/networkfusion/nanoFramework.System.Data.SQLite

It still requires a lot more work before it is ready for integration, but hoping there is community contributions towards it!

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

File: src/nanoFramework.System.Data.Sqlite/nf_system_data_sqlite.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    SQLiteErrorCode_Notice = 27,
+    SQLiteErrorCode_Warning = 28,
+    SQLiteErrorCode_Row = 100,
+    SQLiteErrorCode_Done = 101,
+    SQLiteErrorCode_NonExtendedMask = 255,
+} SQLiteErrorCode;
+
+struct Library_nf_system_data_sqlite_System_Data_Sqlite_ResultSet
+{
+    static const int FIELD__<RowCount>k__BackingField = 1;
```
Comment (josesimoes) [inline src/nanoFramework.System.Data.Sqlite/nf_system_data_sqlite.h:66]: @networkfusion you need to implement the backing fields for these properties.

File: targets/ChibiOS/ST_STM32F769I_DISCOVERY/nanoCLR/nanoFramework.Graphics/DSI_To_Display_Video_Mode.cpp [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 #include "stm32f769xx.h"
 #include "stm32f7xx.h"
-#include "Core_cm7.h"
+#include "core_cm7.h"
```
Comment (josesimoes) [inline targets/ChibiOS/ST_STM32F769I_DISCOVERY/nanoCLR/nanoFramework.Graphics/DSI_To_Display_Video_Mode.cpp:15]: These belong to the other PR, shouldn't be here.

Comment (networkfusion) [reply targets/ChibiOS/ST_STM32F769I_DISCOVERY/nanoCLR/nanoFramework.Graphics/DSI_To_Display_Video_Mode.cpp:15]: I agree, but also need to have a working build. I will merge in the changes once the PR is merged 😉

File: src/nanoFramework.System.Data.Sqlite/nf_system_data_sqlite.h (hunk 1 above)
Comment (networkfusion) [reply src/nanoFramework.System.Data.Sqlite/nf_system_data_sqlite.h:66]: I am quite out of my depth... 😱

Comment (josesimoes) [reply src/nanoFramework.System.Data.Sqlite/nf_system_data_sqlite.h:66]: Oh. Let me help : this is on the C# code. You need to add fields to the class to be the backing fields. We can't have auto-properties in nano when the code is used in native.

Comment (networkfusion) [reply src/nanoFramework.System.Data.Sqlite/nf_system_data_sqlite.h:66]: lol... I will have to revisit, as I am starting to count sheep 🤣

**Pull request closed without merging**